model-zoo/src/main/java/ai/djl/basicmodelzoo/cv/classification/MobileNetV1.java

```diff
@@ -0,0 +1,273 @@
+                                (int) (builder.filters[6] * builder.alpha),
+                                (int) (builder.filters[7] * builder.alpha),
+                                1,
+                                builder))
+                .add(
+                        depthSeparableConv2d(
+                                (int) (builder.filters[6] * builder.alpha),
+                                (int) (builder.filters[7] * builder.alpha),
+                                1,
+                                builder))
+                .add(
+                        depthSeparableConv2d(
+                                (int) (builder.filters[6] * builder.alpha),
+                                (int) (builder.filters[7] * builder.alpha),
+                                1,
+                                builder))
+                .add(
+                        depthSeparableConv2d(
+                                (int) (builder.filters[6] * builder.alpha),
+                                (int) (builder.filters[7] * builder.alpha),
+                                1,
+                                builder))
+                .add(
+                        depthSeparableConv2d(
+                                (int) (builder.filters[6] * builder.alpha),
+                                (int) (builder.filters[7] * builder.alpha),
+                                1,
+                                builder))
+                // separable conv8
+                .add(
+                        depthSeparableConv2d(
+                                (int) (builder.filters[7] * builder.alpha),
+                                (int) (builder.filters[8] * builder.alpha),
+                                2,
+                                builder))
+                // separable conv9
+                .add(
+                        depthSeparableConv2d(
+                                (int) (builder.filters[8] * builder.alpha),
+                                (int) (builder.filters[9] * builder.alpha),
+                                1,
+                                builder)) // maybe the paper goes wrong here
+                // AveragePool
+                .add(Pool.globalAvgPool2dBlock())
+                // FC
+                .add(Linear.builder().setUnits(builder.outSize).build());
+
+        return mobileNet;
+    }
+
+    /**
+     * Creates a builder to build a {@link MobileNetV1}.
+     *
+     * @return a new builder
+     */
+    public static Builder builder() {
+        return new MobileNetV1.Builder();
+    }
+
+    /** The Builder to construct a {@link MobileNetV1} object. */
+    public static final class Builder {
+        float batchNormMomentum = 0.9f;
+        float alpha = 1f; // width multiplier defined in the paper
+
+        long outSize = 10; // 10 as default for basic datasets like cifar10 or mnist
+
+        int[] filters = {32, 64, 128, 128, 256, 256, 512, 512, 1024, 1024};
```

Comment:
Here, you have filters as part of the builder but it isn't actually modifiable. This can be confusing because it is not how the Builder pattern typically works. There really a few straightforward optiosn:

1. Move the option to a hardcoded part of `mobilenet(..)` to more clearly indicate that it is not part of the Builder pattern
2. Create an option to modify `filters`. Then, you would also need to handle the case of different filter lengths. This can be either by adding a validation to require the specific length of 10 or by creating a loop so the block can be built with arbitrary filter lengths. Maybe this would include adding a strides array with the additional restriction that strides and filters have the same length.